configs/terraform/environments/dev/artifact-registry/main.tf

```
@@ -1,14 +1,13 @@
 module "artifact_registry" {
-  source                           = "../../../modules/artifact-registry"
-  for_each                         = toset(var.artifact_registry_names)
-  artifact_registry_name           = each.value
-  artifact_registry_type           = var.artifact_registry_type
-  artifact_registry_module         = var.artifact_registry_module
-  immutable_artifact_registry      = var.immutable_artifact_registry
-  artifact_registry_multi_region   = var.artifact_registry_multi_region
-  artifact_registry_owner          = var.artifact_registry_owner
-  artifact_registry_serviceaccount = var.artifact_registry_serviceaccount
-
-
+  source = "../../../modules/artifact-registry"
 
+  for_each                                 = { for i, v in var.artifact_registry_list : i => v }
```

Can you use a toset here, instead iterating over list.

```suggestion
  for_each                                 = toset(var.artifact_registry_list)
```